dali/operators/generic/slice/slice_attr.h

```diff
@@ -97,21 +132,65 @@ class SliceAttr {
   }
 
  private:
-  template <typename T>
-  void ProcessArgumentsHelper(int data_idx,
-                              const T *slice_anchor_data,
-                              const T *slice_shape_data) {
-    bool normalized_anchor = normalized_anchor_;
-    bool normalized_shape  = normalized_shape_;
-    // If integer anchor and shape were provided, assume absolute coordinates silently
-    if (!std::is_floating_point<T>::value) {
-      normalized_anchor = false;
-      normalized_shape = false;
-    }
-    assert(std::is_floating_point<T>::value || (!normalized_anchor && !normalized_shape));
+  void ProcessNamedArgs(int data_idx) {
+    crop_window_generators_[data_idx] =
+      [this, data_idx](const TensorShape<> &shape, const TensorLayout& shape_layout) {
+        CropWindow slice;
+        slice.anchor = std::vector<int64_t>(shape.size(), 0);
+        slice.shape = shape;
+
+        auto axes = axes_;
+        if (!axis_names_.empty()) {
+          axes = GetDimIndices(shape_layout, axis_names_).to_vector();
+        }
+
+        for (size_t i = 0; i < axes.size(); i++) {
+          auto dim = axes[i];
+
+          double anchor_val = 0;
+          if (start_.IsDefined()) {
+            anchor_val = start_[data_idx].data[i];
+          } else if (rel_start_.IsDefined()) {
+            anchor_val = rel_start_[data_idx].data[i] * shape[dim];
+          }
+
+          double end_val = -1.0;
+          if (end_.IsDefined()) {
+            end_val = end_[data_idx].data[i];
+          } else if (rel_end_.IsDefined()) {
+            end_val = rel_end_[data_idx].data[i] * shape[dim];
+          } else if (shape_.IsDefined()) {
+            double shape_val = shape_[data_idx].data[i];
```

Comment:
Is it validated somewhere? What happens when we feed some crazy values there?
```suggestion
            double shape_val = shape_[data_idx].data[i];
            DALI_ENFORCE(shape_val >= 0 && shape_val < max_value<int64_t>(),
              make_string("Shape extent ", shape_val, " is out of valid range 0..", max_value<int64_t>());
```